frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseOverrideForkPointAction.java

```diff
@@ -0,0 +1,139 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static org.checkerframework.checker.i18nformatter.qual.I18nConversionCategory.GENERAL;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsException;
+import com.intellij.openapi.vfs.VirtualFile;
+import git4idea.config.GitConfigUtil;
+import io.vavr.collection.List;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+
+import com.virtuslab.gitmachete.backend.api.IGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteCommit;
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.OverrideForkPointDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+import com.virtuslab.qual.internal.NotUiThreadSafe;
+
+@CustomLog
+public abstract class BaseOverrideForkPointAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeyProject,
+      IExpectsKeyGitMacheteRepository,
+      ISyncToParentStatusDependentAction {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  public @I18nFormat({}) String getActionNameForDescription() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description-action-name");
+  }
+
+  @Override
+  public @I18nFormat({GENERAL, GENERAL}) String getEnabledDescriptionFormat() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description");
+  }
+
+  @Override
+  public @I18nFormat({}) String getActionTextForCurrentBranch() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.text.current-branch");
+  }
+
+  @Override
+  public List<SyncToParentStatus> getEligibleStatuses() {
+    return List.of(SyncToParentStatus.InSyncButForkPointOff);
+  }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
```

Comment:
Maybe it would be even better to hide this action from context menu instead of disabling it

Reply:
And (I just thinking loud) isn't it better if this action be available just when these two positions (`Parent branch` and `Inferred commit`) are different?

Reply:
Yes, that's exactly what happens - it's only available for yellow-edge branches, i.e. when fork point is NOT equal to parent branch commit :)